### PR TITLE
Web example / Monorepo

### PR DESCRIPTION
**Pull request by eshaz**
> Refactor modules into monorepo
> Adds a demo web example
> Fix metadata parsing issues
> Parse metadata using utf-8

### CODE REVIEW

File: src/demo/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "@testing-library/user-event": "^7.2.1",
+    "gh-pages": "^3.1.0",
+    "prettier": "^2.1.2"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
+    "format": "prettier --write '**/*.js' --write '**/*.json' --write '**/*.jsx'",
```
Comment (eshaz) [inline src/demo/package.json:22]: Need to exclude build folder

File: src/demo/src/App.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+function App() {
+  const [station, setStation] = useState();
+
+  return (
+    <div>
+      <header className={styles.header}>
+        <About />
+      </header>
+      <div className={styles.headerGradient}></div>
```
Comment (eshaz) [inline src/demo/src/App.js:16]: Remove un-used div

File: src/demo/src/icecast/BufferArray.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/**
+ * @description Stores a collection of buffers as an array.
+ */
+export default class BufferArray {
```
Comment (eshaz) [inline src/demo/src/icecast/BufferArray.js:4]: Probably don't need the array logic with the current MetadataPlayer implementation.

File: src/icecast-metadata-js/test/IcecastMetadataReader.test.js [hunk 4]
```diff
@@ -502,6 +504,20 @@ describe("Icecast Metadata Reader", () => {
         expect(returnedMetadata).toEqual(expectedMetadata);
       });
 
+      it("should parse given metadata value contains non alpha numeric characters", () => {
```
Comment (eshaz) [inline src/icecast-metadata-js/test/IcecastMetadataReader.test.js:507]: Should also add a test for unicode

File: src/stream-recorder/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "stream-recorder",
+  "version": "0.0.0",
+  "description": "Icecast stream recorder that saves audio and metadata",
+  "scripts": {
+    "test": "jest",
+    "format": "prettier --write '**/*.js' --write '**/*.json'",
```
Comment (eshaz) [inline src/stream-recorder/package.json:7]: exclude dist folder
